eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -206,39 +206,100 @@ object Transactions {
 
   def decodeTxNumber(sequence: Long, locktime: Long): Long = ((sequence & 0xffffffL) << 24) + (locktime & 0xffffffL)
 
-  def makeCommitTx(commitTxInput: InputInfo, commitTxNumber: Long, localPaymentBasePoint: PublicKey, remotePaymentBasePoint: PublicKey, localIsFunder: Boolean, localDustLimit: Satoshi, localRevocationPubkey: PublicKey, toLocalDelay: CltvExpiryDelta, localDelayedPaymentPubkey: PublicKey, remotePaymentPubkey: PublicKey, localHtlcPubkey: PublicKey, remoteHtlcPubkey: PublicKey, spec: CommitmentSpec): CommitTx = {
+  /**
+   * Represent a link between a commitment spec item (to-local, to-remote, htlc) and the actual output in the commit tx
+
+   * @param output transaction output
+   * @param redeemScript redeem script that matches this output (most of them are p2wsh)
+   * @param specItem commitment spec item this output is built from
+   */
+  case class CommitmentOutputLink(output: TxOut, redeemScript: Seq[ScriptElt], specItem: CommitmentOutput)
+
+  object CommitmentOutputLink {
+    /**
+      * We sort HTLC outputs according to BIP69 + CLTV as tie-breaker for offered HTLC, we do this only for the outgoing
+      * HTLC because we must agree with the remote on the order of HTLC-Timeout transactions even for identical HTLC outputs.
+      * See https://github.com/lightningnetwork/lightning-rfc/issues/448#issuecomment-432074187.
+      */
+    def sort(a: CommitmentOutputLink, b: CommitmentOutputLink): Boolean = (a.specItem, b.specItem) match {
+      case (DirectedHtlc(OUT, htlcA), DirectedHtlc(OUT, htlcB)) if htlcA.paymentHash == htlcB.paymentHash && htlcA.amountMsat == htlcB.amountMsat =>
+        htlcA.cltvExpiry <= htlcB.cltvExpiry
+      case _ => LexicographicalOrdering.isLessThan(a.output, b.output)
+    }
+  }
+
+  def makeCommitTxOutputs(localIsFunder: Boolean,
+                          localDustLimit: Satoshi,
+                          localRevocationPubkey: PublicKey,
+                          toLocalDelay: CltvExpiryDelta,
+                          localDelayedPaymentPubkey: PublicKey,
+                          remotePaymentPubkey: PublicKey,
+                          localHtlcPubkey: PublicKey,
+                          remoteHtlcPubkey: PublicKey,
+                          spec: CommitmentSpec): Seq[CommitmentOutputLink]  = {
```

Comment:
Nit: we may want to define a `type` = `Seq[CommitmentOutputLink]` (or a container class), because the whole point of this is to preserve the order.